lightning/src/util/persist.rs

```diff
@@ -205,3 +213,760 @@ where
 	}
 	Ok(res)
 }
+
+struct MonitorUpdatingPersister<K: Deref, L: Deref>
```

Comment:
Hm, I named it that because it implements `Persist` and is therefore a persister. Were you thinking about relocating the whole-monitor-persist implementation to its own struct, rather than a trait of the KVStore, after some discussion below?